docs/readthedocs-reqs.txt

```diff
@@ -1,3 +1,5 @@
+sphinx >= 4.3.2
+sphinx_rtd_theme >= 1.0.0
```

Comment:
My understanding via the build logs is that RTD already installs the latest versions of these packages before invoking this requirements file. The following is a screenshot of the docs build for this PR from [here](https://readthedocs.org/projects/smqtk-iqr/builds/24332182/):
![image](https://github.com/Kitware/SMQTK-IQR/assets/735270/390daa2a-1657-4871-846d-1a7950dd5580)

Reply:
The `sphinx_rtd_theme` package was not there and it threw an error. In general, I think the docs requirements.txt file should contain the dependencies anyone would need to build the docs. If RTD servers have some of the deps pre-installed, that's fine, it saves us bandwidth and it isn't too much overhead to check. 

I should probably add readthedocs-sphinx-ext as an additional dependency as well. Just to ensure anyone who takes the reasonable steps of installing doc dependencies and then building the docs will get a similar experience to what happens on the RTD servers.

Reply:
The poetry dev-dependencies cover documentation requirements, this file ostensibly is just for RTD. If you want to add all the requirements here for documentation building, I'm not going to get in the way. I just don't realistically expect anyone to use this to install doc dependencies separate from a `poetry install`.

Reply:
It would be nice if there was a single source of truth. Is it possible to point .readthedocs at a particular "extras" in the pyproject.toml? Possible lead: https://github.com/readthedocs/readthedocs.org/issues/4912

It also bugs me that the docs dependencies are mixed in with the dev dependencies in the pyproject.toml. Can this be refactored into something like;

```
[tool.poetry.dev-dependencies]
Sphinx = "^3.5.3"
sphinx-rtd-theme = "^0.5.1"
sphinx-argparse = "^0.2.5"
sphinx-prompt = "^1.4.0"

[tool.poetry.extras]
docs = ["sphinx-rtd-theme", "sphinx", "sphinx-argparse", "sphinx-prompt"]
```

And then do something like this in `.readthedocs`:

```
  jobs:
    post_create_environment:
      # Install poetry
      # https://python-poetry.org/docs/#installing-manually
      - python -m pip install poetry
    post_install:
      # Install dependencies with 'docs' dependency group
      # https://python-poetry.org/docs/managing-dependencies/#dependency-groups
      - VIRTUAL_ENV=$READTHEDOCS_VIRTUALENV_PATH poetry install --with docs
```

?

Reply:
FYI: I'm also fine with merging as-is. While it's annoying the deps are specified twice, it's not that hard of a thing to work with. We could easily punt on this as a TODO.

Reply:
I'm not opposed with updating the RTD build to use poetry, but IIRC there are some subtle gotchas with what you posted directly. It would just need some attention and iteration to get it working correctly. I'm fine with where this branch is currently for the time being.